File: analysis/import/SonarImporter/src/main/java/de/maibornwolff/codecharta/importer/sonar/SonarCodeURLLinker.java
package de.maibornwolff.codecharta.importer.sonar;

import de.maibornwolff.codecharta.importer.sonar.model.Component;

import java.net.URL;

public class SonarCodeURLLinker {

    private final String baseCodeUrl;

    public static final SonarCodeURLLinker NULL = new SonarCodeURLLinker() {
        public String createUrlString(Component c) {
            return "";
        }
    };

    private SonarCodeURLLinker() {
        baseCodeUrl = "";
    }

    public SonarCodeURLLinker(URL baseUrlFrom) {
        baseCodeUrl = baseUrlFrom + "/code?id=";
    }

    public String createUrlString(Component component) {
        return baseCodeUrl + component.getKey();
    }
}

File: analysis/import/SonarImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/sonar/SonarCodeURLLinker.kt
package de.maibornwolff.codecharta.importer.sonar

import de.maibornwolff.codecharta.importer.sonar.model.Component

import java.net.URL

open class SonarCodeURLLinker {

    private val baseCodeUrl: String

    private constructor() {
        baseCodeUrl = ""
    }

    constructor(baseUrlFrom: URL?) {
        baseCodeUrl = baseUrlFrom.toString() + "/code?id="
    }

    open fun createUrlString(component: Component): String {
        return baseCodeUrl + component.key!!
    }

    companion object {

        val NULL: SonarCodeURLLinker = object: SonarCodeURLLinker() {
            override fun createUrlString(c: Component): String {
                return ""
            }
        }
    }
}
